### PR TITLE
feat: add segments to "External Recommended Readings"

### LINKED ISSUE #1550: Learning Path LP: Add "segment" to "External recommended readings"

### PR DESCRIPTION
**Pull request by eddybrando**
Closes #1550

### CODE REVIEW

**Comment by eddybrando** [comment]
@y4izus rebase cleanup is done 🙂

File: components/textbook-demo/ExternalRecommendedReadings.vue [hunk 1]
```diff
@@ -5,22 +5,24 @@
...
 
 <script lang="ts">
 import Vue from 'vue'
 import { Component, Prop } from 'vue-property-decorator'
+import { SegmentData } from '~/constants/appLinks'
 
 @Component
 export default class ExternalRecommendedReadings extends Vue {
-  @Prop({ type: Array, required: true }) links!: string[]
+  @Prop({ type: Array, required: true }) links!: SegmentData[]
```
Comment (y4izus) [inline components/textbook-demo/ExternalRecommendedReadings.vue:25]: Should be a GeneralLink, not a SegmentData type

File: constants/appLinks.ts [hunk 2]
```diff
@@ -77,6 +77,23 @@ const TEXTBOOK_DEMO_START_LEARNING: GeneralLink = {
   }
 }
 
+const TEXTBOOK_DEMO_INTRODUCTION_COURSE_EXTERNAL_RECOMMENDED_READINGS: GeneralLink[] = [
```
Comment (y4izus) [inline constants/appLinks.ts:80]: Please, move this definition to the `introduction-couse.vue` page

**Review by y4izus — CHANGES_REQUESTED** [review]
Thank you! just a couple of modifications :)

File: components/textbook-demo/ExternalRecommendedReadings.vue (hunk 1 above)
Comment (eddybrando) [reply components/textbook-demo/ExternalRecommendedReadings.vue:25]: Good catch! Thanks a lot 🙂

File: constants/appLinks.ts (hunk 2 above)
Comment (eddybrando) [reply constants/appLinks.ts]: Sure, as discussed offline, we'll start moving the links to the components that use them. 👌

**Comment by eddybrando** [comment]
@y4izus thanks a lot for the review. I've implemented the changes requested. 🙂

File: pages/textbook-demo/learning-paths/introduction-course.vue [hunk 3]
```diff
@@ -10,6 +10,9 @@
 <script lang="ts">
 import { Component } from 'vue-property-decorator'
 import QiskitPage from '~/components/logic/QiskitPage.vue'
+import { GeneralLink } from '~/constants/appLinks'
+
+const segmentActionBase = 'introduction-course > external-recommended-readings >'
```
Comment (y4izus) [inline pages/textbook-demo/learning-paths/introduction-course.vue:15]: If you move this line after [line 26](https://github.com/Qiskit/qiskit.org/pull/1563/files#diff-c0470fa3ad319c3ec454915a8dae5f0f5bc2def97b395e1078e3329dea09756aR26), you can reuse the `routeName` and we have the constant close to where it is used

**Review by y4izus — CHANGES_REQUESTED** [review]
Thanks!! just one more change and we are done ;)

File: pages/textbook-demo/learning-paths/introduction-course.vue (hunk 3 above)
Comment (eddybrando) [reply pages/textbook-demo/learning-paths/introduction-course.vue]: Good idea! I'll change it. 🙂

File: pages/textbook-demo/learning-paths/introduction-course.vue [hunk 4]
```diff
@@ -21,9 +22,24 @@ import QiskitPage from '~/components/logic/QiskitPage.vue'
...
-    'https://machinelearningmastery.com/gentle-introduction-linear-algebra/'
+
+  segmentActionBase = `${this.routeName} > external-recommended-readings >`
+
+  links: GeneralLink[] = [
+    {
+      url: 'https://math.mit.edu/~gs/linearalgebra/',
+      label: 'https://math.mit.edu/~gs/linearalgebra/',
+      segment: {
+        action: `${this.segmentActionBase}  mit-linear-algebra`
```
Comment (eddybrando) [inline pages/textbook-demo/learning-paths/introduction-course.vue:33]:
```suggestion
        action: `${this.segmentActionBase} mit-linear-algebra`
```

**Review by y4izus — APPROVED** [review]
Great! Thank you Eddy!

**Pull request merged**